src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -1783,26 +1783,32 @@ _find_topology_version (const bson_t *reply, bson_t *topology_version)
  * Caller must lock topology->mutex. */
 void
 _mongoc_topology_clear_connection_pool (mongoc_topology_t *topology,
-                                        uint32_t server_id)
+                                        uint32_t server_id,
+                                        const bson_oid_t *service_id)
 {
    mongoc_server_description_t *sd;
    bson_error_t error;
 
+   BSON_ASSERT (service_id);
+
    sd = mongoc_topology_description_server_by_id (
       &topology->description, server_id, &error);
    if (!sd) {
       /* Server removed, ignore and ignore error. */
       return;
    }
-   TRACE ("clearing pool for server: %s", sd->host.host_and_port);
```

Comment:
Why remove the `TRACE`? I see this was moved to `_mongoc_topology_get_connection_pool_generation` but I'm not sure I understand why.